download_geo_files.py

```diff
@@ -113,18 +85,34 @@ def download_supplementary_files(accession_id):
             logging.info('Skipping existing file: %s', save_file_full)
         else:
             logging.info('Downloading to: %s', save_file_full)
-            headers = download_file(url, save_file_full)
-            if not file_name_found:
-                # try one last time to find filename in headers and rename downloaded file if found
-                save_file_name = filename_from_headers(headers)
-                if save_file_name:
-                    new_file_full = f'{save_file_path}/{save_file_name}'
-                    if not os.path.isfile(new_file_full):
-                        os.rename(save_file_full, new_file_full)
-                        logging.info('Renaming %s to %s', save_file_full, new_file_full)
+            download_file(url, save_file_full)
     return save_file_path
 
 
+def supplementary_file_download_links(html: str) -> Sequence:
+    """
+    Locate the Supplementary file table in the given html
+    and pluck out the file name & urls to download the file
+    """
+    source = furl.furl(source_url_template)
+    match = re.search(r'<table(?:(?!table).)+?>Supplementary file<(?:(?!table).)+</table>', html, re.DOTALL)
+    if match:
+        # For each row in the table get the filename and the http link url
+        table_html = match.group()
+        match = re.findall(r'<tr[^>]*>'                             # tr up to first td
```

Comment:
For future reference: https://www.crummy.com/software/BeautifulSoup/bs4/doc/ is very handy for parsing scraped HTML.